src/model.ts

```diff
@@ -12,6 +12,23 @@ export const createDefaultValidationResult = (): ValidationResult => ({
   message: '',
 });
 
+export interface FormFieldError {
+  key: string;
+  validationResult: ValidationResult;
+}
+
+export interface FormValidationSummary {
+  succeeded: boolean;
+  fieldErrors: FormFieldError[];
```

Comment:
@nasdan FormValidationSummary will return an array, by following this approach we remove the issue of nested field creating objects, maybe we can remove the dependency with lodash.set

Reply:
This was our first approach, returning an array. And we moved in to `lodash.set` one because we want use the result like:

```
validateForm(vm).then(formResult => {
   setLoginErrors({
      ...loginErrors,
      ...formResult.fieldErrors,
   })
})
```
If we return an array, where we use it, we always have to create the `reduce` function.

Reply:
Good point, not sure if we can just build a helper for this.

Or if we can find a solution for the _set_ and _nested_ fields.

I think we can keep it like this now, but once we start building examples check what can be easier: 

 - Add a helper to do this dirty job.
 - Refactor to object {key, value} but checking how to solve the nested properties issue

Reply:
Ok, if you want I could create a branch resolving nested properties

Reply:
I will create the branch in lcFormValidation